### ISSUE TITLE
Changed row height from "wordwrap" isn't applied in side by side view

### ISSUE THREAD

**Issue by codingtwinky** [rtfpessoa/diff2html#99]
Thank for a quick response on #97 and we do have word wrap working in default view but not in side by side view.  Wordwrap view has always been challenging in side by side view but I figure I would use this time to describe the problem. 

Side by side view with wordwrap is challenging because when word is wrapped in one table, that height needs to be applied to another table.  For example, in below right table's words are wrapped and caused increase in row height but it cannot be easily be applied to left table as they are separate table.  

<img width="984" alt="screen shot 2016-09-11 at 15 06 32" src="https://cloud.githubusercontent.com/assets/5281068/18420886/39783978-7832-11e6-839a-a2481e0ee668.png">
 
Will it be easy to place them in a same table so we can easily do this without JS?  I feel like this will not be a trivial change and I think this maybe not be a critical issue.

**Comment by rtfpessoa** [comment]
That is probably the biggest layout problem.
Even for the line-by-line view it hard to get the breaks correct.

For the side-by-side having only one table. I am not sure if you can make the widths and the overflows work. And even harder should be to have scroll.

If you have any suggestions. I can take a look, but this is quite difficult.

**Comment by rtfpessoa** [comment]
Related to #86

**Comment by thinkpozzitive** [comment]
@codingtwinky what changes did you make in order to get "word wrap working in default view"? Can you show me an example?

**Comment by mohanapreeya** [comment]
I have solved this issue in my worspace with some new template files and code changes. I would like to contribute my changes.

**Comment by rtfpessoa** [comment]
@mohanapreeya that would be nice. Can I check the working version somewhere?

**Comment by mohanapreeya** [comment]
I have a temporary output in this fork https://github.com/mohanapreeya/diff2html.git. You can view the output in demo page. Source code changes are in the same repository.

**Comment by hungdev** [comment]
@jung-kim could you tell me how to break word like your image?

**Comment by hungdev** [comment]
@jung-kim yes, which class should i override `overflow-wrap: break-word;`?
or how do i do to make like your image?

**Comment by hungdev** [comment]
@jung-kim no, my mean is break the work if the sentence too long.
look at the online example: https://stackblitz.com/edit/react-em2guv
i dont want to horizontal scroll, i want to break line.

**Comment by zabbixfan** [comment]
I really need this feature,please

**Comment by robertwalsh0** [comment]
Not sure if this works in side-by-side but in line-by-line you can get the lines to wrap by changing the following css

```
    .d2h-code-line, .d2h-code-line-ctn {
      white-space: normal;
    }
```

**Comment by ccadek** [comment]
@robertwalsh0 This CSS does work in side-by-side view as well.

**Comment by rtfpessoa** [comment]
I cannot replicate this, can you provide a working example?

**Comment by NL33** [comment]
Is there currently a working way to have text wrap with any of diff2html's formats?

**Comment by rtfpessoa** [comment]
> Is there currently a working way to have text wrap with any of diff2html's formats? 

No, I could not find a way that does not break something else.

**Comment by mensfeld** [comment]
@rtfpessoa could you point out what were the problems, I did really bad POC but this looks sooo good (not side by side):

![Zrzut ekranu z 2021-09-17 13-05-01](https://user-images.githubusercontent.com/392754/133772720-cdf4914a-b6de-4944-a6de-24a7219c52ee.png)

**Comment by rtfpessoa** [comment]
@mensfeld the problem is in the side-by-side diff. Can you make that work for side-by-side too?

**Comment by mensfeld** [comment]
@rtfpessoa I can give it a shot in 1-2 weeks but I'm not an expert and my solution is rather clumsy.

**Comment by donaldpipowitch** [comment]
> Not sure if this works in side-by-side but in line-by-line you can get the lines to wrap by changing the following css
> 
> ```
>     .d2h-code-line, .d2h-code-line-ctn {
>       white-space: normal;
>     }
> ```

Thanks for the hint. I got even better results with this as leading whitespace in a line will be preserved:

```
     .d2h-code-line, .d2h-code-line-ctn {
       white-space: pre-wrap;
     }
```

**Comment by skivol** [comment]
Taking ideas from this thread (thanks for those :) ) and also overriding original "width: 100%" in some classes (which probably doesn't work well inside table cells, in combination with padding):
```css
.d2h-code-line-ctn {
	white-space: normal;
}
.d2h-code-line, .d2h-code-side-line, .d2h-code-line-ctn {
	width: revert;
}
```
to avoid overflow problems like on the image below:
![example](https://user-images.githubusercontent.com/2342155/163843850-68f7390c-ab90-4881-bdc5-59d4cb82afa6.png)

And things seem to look quite good 👍

**Comment by rtfpessoa** [comment]
@skivol do you think you could send a full spec with all the changes required to get this to work in bot side-by-side and line-by-line?

**Comment by skivol** [comment]
@rtfpessoa you mean the full source code of the html template which I used for the purpose (with css styles which I mentioned in the comment above) ?
Can do that, but not sure if the quality of the results is sufficient for general usage (specifically in regards to the alignment of the lines to the left and right which is a bit worse when wrapping the lines, but the benefit is that everything is visible without the need to scroll horizontally).

![image](https://user-images.githubusercontent.com/2342155/167311478-dc245a6f-ebc1-45c0-b966-b737efebdd44.png)

I guess one could try to display both sides in one table (possibly with rowspan for changesets which differ in number of rows) in an attempt to be able to do the wrapping and keep the alignment 🤔

**Comment by rtfpessoa** [comment]
Ahh, ok. That not needed

**Comment by abelpz** [comment]
@rtfpessoa is there an open PR on this? I got it working with some CSS changes.

**Comment by rtfpessoa** [comment]
👋 @abelpz no, I could never get it working properly.

If you got it working in both ways, would be good if we could offer both options (with and without line breaks) maybe by applying a different css classes.

**Comment by abelpz** [comment]
@rtfpessoa it does work for both side-by-side and line-by-line, I could try to add that if you want, this is what I propose:

- Add one config option to set the default mode: `wordWrap: true`
- Add a button somewhere in the UI to toggle this feature.

If you point me to the files I should look at to add this, I can make a PR. Or I could just share the CSS, and someone else can give it a try.

**Comment by rtfpessoa** [comment]
I am not sure this can be done with a single file change.

The code starts in [diff2html.ts](https://github.com/rtfpessoa/diff2html/blob/master/src/diff2html.ts).
You need to add the css in [diff2html.css](https://github.com/rtfpessoa/diff2html/blob/master/src/ui/css/diff2html.css) and then somehow apply the class change when rendering maybe by passing the boolean to the templates.

You can test if it works by running `yarn validate` and then opening the `docs/demo.html` file generated which will use the code you changed.

**Comment by jsonmorris** [comment]
> Or I could just share the CSS, and someone else can give it a try.

@abelpz any way you could provide the CSS for definition the table width and word-wrap correctly for Side-by-side diffs?

**Comment by jsonmorris** [comment]
I achieved the wrap I needed for side-by-side diffs using the following CSS. Not sure if this breaks line-by-line, but looks like there are solutions for that above.

```
.d2h-code-line-ctn {
    white-space: pre-wrap;
    word-break: break-all;
}
```

**Comment by hayatogh** [comment]
Hi! First off, thank you for all the great work on this project.

I've started working on this.

My primary motivation is to generate diffs for static environments like email or chat, where HTML renders but JavaScript is often disabled, making synchronized scrolling unusable. A wrapped side-by-side view would be perfect for this use case.

I will make a PR soon but due to internal HTML structure changes, it will not be just a checkbox option, but the third output mode.
It was much easier for me to implement.  It may not be an acceptable changes but even in that case, I hopefully it will be a good starting point at least.
